app/Http/Requests/Album/DeleteTrackRequest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Http\Requests\Album;
+
+use App\Http\Requests\BaseApiRequest;
+use App\Http\Requests\Contracts\HasAbstractAlbum;
+use App\Http\Requests\Contracts\HasAlbum;
+use App\Http\Requests\Traits\HasAlbumTrait;
+use App\Rules\AlbumIDRule;
+
+/**
+ * @implements HasAlbum<AbstractAlbum>
+ */
+class DeleteTrackRequest extends BaseApiRequest implements HasAlbum
+{
+	use HasAlbumTrait;
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function authorize(): bool
+	{
+		return $this->authorizeAlbumWrite($this->album);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function rules(): array
+	{
+		return [
+			HasAbstractAlbum::ALBUM_ID_ATTRIBUTE => ['present', new AlbumIDRule(false)],
+		];
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function processValidatedValues(array $values, array $files): void
+	{
+		$this->album = $this->albumFactory->findAlbumOrFail(
+			$values[HasAbstractAlbum::ALBUM_ID_ATTRIBUTE], false
+		);
```

Comment:
```suggestion
		$this->album = Album::query()->findOrFail($values[HasAbstractAlbum::ALBUM_ID_ATTRIBUTE]);
```

suffices. No need for the added factory method. The only reason why I had to added a factory method for `BaseAlbum` (the one which you copied) was that a `BaseAlbum` can either be an `Album` or a `TagAlbum`. But as tracks are only supported for real albums, a simply Eloquent query suffices.